library/azure_rm_devtestlabslab.py

```diff
@@ -0,0 +1,362 @@
+                response = self.get_poller_result(response)
+
+        except CloudError as exc:
+            self.log('Error attempting to create the Lab instance.')
+            self.fail("Error creating the Lab instance: {0}".format(str(exc)))
+        return response.as_dict()
+
+    def delete_lab(self):
+        '''
+        Deletes specified Lab instance in the specified subscription and resource group.
+
+        :return: True
+        '''
+        self.log("Deleting the Lab instance {0}".format(self.name))
+        try:
+            response = self.mgmt_client.labs.delete(resource_group_name=self.resource_group,
+                                                    name=self.name)
+        except CloudError as e:
+            self.log('Error attempting to delete the Lab instance.')
+            self.fail("Error deleting the Lab instance: {0}".format(str(e)))
+
+        return True
+
+    def get_lab(self):
+        '''
+        Gets the properties of the specified Lab.
+
+        :return: deserialized Lab instance state dictionary
+        '''
+        self.log("Checking if the Lab instance {0} is present".format(self.name))
+        found = False
+        try:
+            response = self.mgmt_client.labs.get(resource_group_name=self.resource_group,
+                                                 name=self.name)
+            found = True
+            self.log("Response : {0}".format(response))
+            self.log("Lab instance : {0} found".format(response.name))
+        except CloudError as e:
+            self.log('Did not find the Lab instance.')
+        if found is True:
+            return response.as_dict()
+
+        return False
+
+    def format_item(self, d):
+        d = {
+            'id': d.get('id', None)
+        }
+        return d
+
+
+def default_compare(new, old, path):
+    if new is None:
+        return True
+    elif isinstance(new, dict):
+        if not isinstance(old, dict):
+            return False
+        for k in new.keys():
+            if not default_compare(new.get(k), old.get(k, None), path + '/' + k):
+                return False
+        return True
+    elif isinstance(new, list):
+        if not isinstance(old, list) or len(new) != len(old):
+            return False
+        if isinstance(old[0], dict):
+            key = None
+            if 'id' in old[0] and 'id' in new[0]:
+                key = 'id'
+            elif 'name' in old[0] and 'name' in new[0]:
+                key = 'name'
+            new = sorted(new, key=lambda x: x.get(key, None))
+            old = sorted(old, key=lambda x: x.get(key, None))
+        else:
+            new = sorted(new)
+            old = sorted(old)
+        for i in range(len(new)):
+            if not default_compare(new[i], old[i], path + '/*'):
+                return False
+        return True
+    else:
+        return new == old
+
+
+def expand(d, path, **kwargs):
```

Comment:
this is really hard to maintain, it's overkill in this module where parameter is quite simple

Reply:
yes, so maybe let's just place it in common, I have tested it in around 60 modules now

Reply:
also added documentation in cosmos db pr

Reply:
not necessary in this module

Reply:
it's removed now